sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -111,8 +110,11 @@ class Column(protected[sql] val expr: Expression) extends Logging {
     // Leave an unaliased generator with an empty list of names since the analyzer will generate
     // the correct defaults after the nested expression's type has been resolved.
     case explode: Explode => MultiAlias(explode, Nil)
+
     case jt: JsonTuple => MultiAlias(jt, Nil)
 
+    case func: UnresolvedFunction => UnresolvedAlias(func)
```

Comment:
I was exactly thinking the same thing (when i had posted my last question) @cloud-fan :-). Thank you. I will make the change.